CHANGELOG.md

```diff
@@ -10,6 +10,8 @@
 
 ### Bug fixes
 
+* `skbio.tree.TreeNode.prune` was not handling a situation in which a parent was validly removed during pruning operations as may happen if the resulting subtree does not include the root.
```

Comment:
Can you please note the previous behavior (silent bug with incorrect results, or errors noisily?)